File: js_sdk/src/model/BcadvertisercreateBillingInfo.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';

/**
 * The BcadvertisercreateBillingInfo model module.
 * @module model/BcadvertisercreateBillingInfo
 * @version 0.1.8
 */
export class BcadvertisercreateBillingInfo {
  /**
   * Constructs a new <code>BcadvertisercreateBillingInfo</code>.
   * Billing information. Required if the place of registration of the ad account or the agency that owns the Business Center is France, Brazil, or Mexico
   * @alias module:model/BcadvertisercreateBillingInfo
   * @class
   */
  constructor() {
  }

  /**
   * Constructs a <code>BcadvertisercreateBillingInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/BcadvertisercreateBillingInfo} obj Optional instance to populate.
   * @return {module:model/BcadvertisercreateBillingInfo} The populated <code>BcadvertisercreateBillingInfo</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new BcadvertisercreateBillingInfo();
      if (data.hasOwnProperty('address'))
        obj.address = ApiClient.convertToType(data['address'], 'String');
      if (data.hasOwnProperty('tax_map'))
        obj.tax_map = ApiClient.convertToType(data['tax_map'], 'String');
    }
    return obj;
  }
}

/**
 * Billing address, no more than 512 characters in length
 * @member {String} address
 */
BcadvertisercreateBillingInfo.prototype.address = undefined;

/**
 * Billing and invoicing tax number. Different countries use different tax number fields. The tax number field `vat` is used for European countries, for instance, France.The tax number field `tax_id` is used for Brazil (required), Mexico (required), United Arab Emirates, Egypt, Saudi Arabia, Israel, Turkey, Canada, and the United States.The tax number field `abn` is used for Australia and New Zealand.
 * @member {String} tax_map
 */
BcadvertisercreateBillingInfo.prototype.tax_map = undefined;

File: java_sdk/src/main/java/business_api_client/BcadvertisercreateBillingInfo.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
/**
 * Billing information. Required if the place of registration of the ad account or the agency that owns the Business Center is France, Brazil, or Mexico
 */
@Schema(description = "Billing information. Required if the place of registration of the ad account or the agency that owns the Business Center is France, Brazil, or Mexico")
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2026-01-08T15:25:30.628143-08:00[America/Los_Angeles]")
public class BcadvertisercreateBillingInfo {
  @JsonProperty("address")
  private String address = null;

  @JsonProperty("tax_map")
  private String taxMap = null;

  public BcadvertisercreateBillingInfo address(String address) {
    this.address = address;
    return this;
  }

   /**
   * Billing address, no more than 512 characters in length
   * @return address
  **/
  @Schema(description = "Billing address, no more than 512 characters in length")
  public String getAddress() {
    return address;
  }

  public void setAddress(String address) {
    this.address = address;
  }

  public BcadvertisercreateBillingInfo taxMap(String taxMap) {
    this.taxMap = taxMap;
    return this;
  }

   /**
   * Billing and invoicing tax number. Different countries use different tax number fields. The tax number field &#x60;vat&#x60; is used for European countries, for instance, France.The tax number field &#x60;tax_id&#x60; is used for Brazil (required), Mexico (required), United Arab Emirates, Egypt, Saudi Arabia, Israel, Turkey, Canada, and the United States.The tax number field &#x60;abn&#x60; is used for Australia and New Zealand.
   * @return taxMap
  **/
  @Schema(description = "Billing and invoicing tax number. Different countries use different tax number fields. The tax number field `vat` is used for European countries, for instance, France.The tax number field `tax_id` is used for Brazil (required), Mexico (required), United Arab Emirates, Egypt, Saudi Arabia, Israel, Turkey, Canada, and the United States.The tax number field `abn` is used for Australia and New Zealand.")
  public String getTaxMap() {
    return taxMap;
  }

  public void setTaxMap(String taxMap) {
    this.taxMap = taxMap;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    BcadvertisercreateBillingInfo bcadvertisercreateBillingInfo = (BcadvertisercreateBillingInfo) o;
    return Objects.equals(this.address, bcadvertisercreateBillingInfo.address) &&
        Objects.equals(this.taxMap, bcadvertisercreateBillingInfo.taxMap);
  }

  @Override
  public int hashCode() {
    return Objects.hash(address, taxMap);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class BcadvertisercreateBillingInfo {\n");
    
    sb.append("    address: ").append(toIndentedString(address)).append("\n");
    sb.append("    taxMap: ").append(toIndentedString(taxMap)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
